plugins/woocommerce-blocks/assets/js/base/utils/use-return-focus.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useCallback } from '@wordpress/element';
+
+let origin: Element | null = null;
+
+/**
+ * Adds the unmount behavior of returning focus to the element which had it
+ * previously as is expected for roles like menus or dialogs.
+ *
+ * This function is copied from Gutenberg's hook under the same name.
+ */
+export function useFocusReturn(
+	onFocusReturn?: () => void
+): ( node: HTMLElement | null ) => void {
+	const ref = useRef< HTMLElement >( null );
+	const focusedBeforeMount = useRef< Element >( null );
+	const onFocusReturnRef = useRef( onFocusReturn );
+	useEffect( () => {
+		onFocusReturnRef.current = onFocusReturn;
+	}, [ onFocusReturn ] );
+
+	return useCallback( ( node ) => {
+		if ( node ) {
+			// Set ref to be used when unmounting.
+			ref.current = node;
+
+			// Only set when the node mounts.
+			if ( focusedBeforeMount.current ) {
+				return;
+			}
+
+			focusedBeforeMount.current = node.ownerDocument.activeElement;
+		} else if ( focusedBeforeMount.current ) {
+			const isFocused = ref.current?.contains(
+				ref.current?.ownerDocument.activeElement
+			);
+
+			if ( ref.current?.isConnected && ! isFocused ) {
```

Comment:
If I see this correctly, uncommenting the `return;` is the only difference between this customised version and [Gutenberg's custom hook](https://github.com/WordPress/gutenberg/blob/0a4c5c2f38d801eedd0de2b6e54258e687442777/packages/compose/src/hooks/use-focus-return/index.js#L31-L78), isn't it? 

In the PR description, you mentioned:
> The last one was tricky because we have the logic to return, it just never triggers because we don't unmount our drawer, we just hide it.

Is there something stopping us from unmounting the drawer instead of hiding it so that we can use the custom hook from Gutenberg, before overwriting it?

Reply:
> Is there something stopping us from unmounting the drawer instead of hiding it so that we can use the custom hook from Gutenberg, before overwriting it?

Yes, the size of the refactor is not worth it and we don't know the risks of doing that. I chatted with @Aljullu around this (given he was the one who introduced this to Mini Cart) and we concluded that we will seek a solution upstream first before rewriting because this used to work fine.

We would also need to rewrite the shipping calculator to make it unmount and return focus.

So the cost of copy pasting a hook is lower than a refactor. Keep also in mind that the Drawer is copy-paste from the Modal component in Gutenberg, with modification.

Reply:
Thanks for clarifying this, @senadir. In that case, it makes sense to overwrite the custom hook instead of refactor the Mini Cart and the Shipping Calculator in our end.